app/controllers/shopping_list_controller.rb

```diff
@@ -0,0 +1,11 @@
+class ShoppingListController < ApplicationController
+  def index
+    @recipe = Recipe.find(params[:recipe_id])
+    @recipe_foods = RecipeFood.where(recipe_id: @recipe.id).includes(:food)
+    @recipe_foods_with_names = @recipe_foods.map do |rf|
+      { id: rf.id, quantity: rf.quantity, recipe_id: rf.recipe_id, food_id: rf.food_id, name: rf.food.name,
+        price: rf.food.price }
+    end
+    @general_shopping_list, @total = current_user.general_shopping_list(@recipe_foods_with_names)
+  end
+end
```

Comment:
- [ ] Please make sure that the lists of the general shopping list are foods that are missing for all recipes of the logged-in user. and should count the total food items and the total price of the missing food. :pray: 

- `HINT`: **Compare the list of food for all recipes with the general food list of that user**

  _As you can see from the video below :point_down:  there are foods that are missing from all recipes but your app is still saying that  `You don't have to buy anything!!` so make you adjust your code to generate all the missing foods as a shopping list_ 
      
    [Screencast from 03-17-2023 09:13:30 PM.webm](https://user-images.githubusercontent.com/92303891/226057543-9cd283a0-4973-452a-ba0b-31028c44c6de.webm)

Reply:
Hello there, for this one, it is only show if your food quantity is less than recipe quantity, @Owusu-Desmond

Reply:
For this one, it happens when user food quantity is smaller than the recipe food quantity.
Suppose we have two oranges in food. WE need 4 oranges to recipe, we will output the calculated food price and 2 oranges will be in there. @Owusu-Desmond

Reply:
I previously created two oranges quantity in food so the recipe is needed 4444 oranges quantity in below pic

![Screenshot from 2023-03-18 04-24-59](https://user-images.githubusercontent.com/105173981/226060148-65dc10fb-22e2-4310-993a-16e9133f0d4f.png)


so when we generates here, it will be 4442. Is it right? if my thinking or understanding to requirements wrong, please comment it back



![Screenshot from 2023-03-18 04-25-14](https://user-images.githubusercontent.com/105173981/226060138-cfa71397-6e44-475f-b8f3-54251a5e5026.png)